src/mscorlib/mscorlib.csproj

```diff
@@ -171,6 +171,16 @@
       <ResGenDefines>$(DefineConstants)</ResGenDefines>
     </SplitTextStringResource>
   </ItemGroup>
+
+  <Target Name="CDefineChecker" BeforeTargets="Build">
+    <!-- Compiler Definition Verification -->
+    <Message Importance="High" Text="============" />
+    <Message Importance="High" Text="DefineConstants = $(DefineConstants)" />
+    <Message Importance="High" Text='Executing $(MSBuildThisFileDirectory)../../scripts/check-definitions.pl $(MSBuildThisFileDirectory)../../cmake.definitions "$(DefineConstants)"' />
+    <Exec Command='perl $(MSBuildThisFileDirectory)../../scripts/check-definitions.pl $(MSBuildThisFileDirectory)../../cmake.definitions "$(DefineConstants)"' />
+    <Message Importance="High" Text="============" />
```

Comment:
I think right now we do not have a dependency on perl. This add new dependency on perl which is not installed by default on windows machines